lib/attributor/hash_dsl_compiler.rb

```diff
@@ -0,0 +1,141 @@
+require_relative 'dsl_compiler'
+
+
+module Attributor
+
+  class HashDSLCompiler < DSLCompiler
+
+    # A class that encapsulates the definition of a requirement for Hash attributes
+    # It implements the validation against incoming values and it describes its format for documentation purposes
+    class Requirement
+      attr_reader :type
+      attr_reader :number
+      attr_reader :attr_names
+      attr_reader :description
+
+      def initialize(spec)
```

Comment:
could make this signature `description: nil, **spec` to avoid the delete below and modifying the incoming hash.